src/components/BridgeForm.js

```diff
@@ -0,0 +1,139 @@
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React, { useState, useEffect } from 'react';
+import { Button, Checkbox, Form, FormGroup, Modal, ModalVariant, TextInput } from '@patternfly/react-core';
+import cockpit from 'cockpit';
+import { useNetworkDispatch, useNetworkState, actionTypes } from '../NetworkContext';
+import interfaceType from '../lib/model/interfaceType';
+
+const _ = cockpit.gettext;
+
+const BridgeForm = ({ isOpen, onClose, bridge }) => {
+    const isEditing = !!bridge;
+    const [name, setName] = useState(bridge?.name || "");
+    const [selectedInterfaces, setSelectedInterfaces] = useState(bridge?.interfaces || []);
+    const [candidateInterfaces, setCandidateInterfaces] = useState([]);
+    const { interfaces } = useNetworkState();
+    const dispatch = useNetworkDispatch();
+
+    useEffect(() => {
+        if (isEditing) {
+            setCandidateInterfaces(Object.values(interfaces).filter(i => i.id !== bridge.id));
+        } else {
+            setCandidateInterfaces(Object.values(interfaces));
+        }
+    }, [interfaces]);
+
+    const addConnection = () => {
+        dispatch({
+            type: actionTypes.ADD_CONNECTION,
+            payload: { name, interfaces: selectedInterfaces, type: interfaceType.BRIDGE }
+        });
+        resetForm();
+    };
+
+    const updateConnection = () => {
+        dispatch({
+            type: actionTypes.UPDATE_CONNECTION,
+            payload: { id: bridge.id, changes: { name, interfaces: selectedInterfaces } }
+        });
+    }
+
+    const addOrUpdateConnection = () => {
+        if (isEditing) {
+            updateConnection();
+        } else {
+            addConnection();
+        }
+        onClose();
+    }
+
+    const closeForm = () => {
+        resetForm();
+        onClose();
+    };
+
+    const resetForm = () => {
+        setName(bridge?.name || "");
+        setSelectedInterfaces(bridge?.interfaces || []);
+    };
+
+    const handleSelectedInterfaces = (name) => (value) => {
+        if (value) {
+            setSelectedInterfaces([...selectedInterfaces, name]);
+        } else {
+            setSelectedInterfaces(selectedInterfaces.filter(i => i !== name));
+        }
+    };
+
+    const isIncomplete = () => {
+        return (name === "" || selectedInterfaces.length === 0);
+    };
+
+    return (
+        <Modal
+            variant={ModalVariant.small}
+            title={isEditing ? _("Edit Bridge") : _("Add Bridge")}
+            isOpen={isOpen}
```

Comment:
We have to pass the `onClose` prop too. Or use the `showClose={false}` instead, which removes the `X` icon from the top-right corner plus avoid using the <kbd>Esc</kbd> key for closing the dialog.

```suggestion
            isOpen={isOpen}
            onClose={closeForm}
```